### ISSUE TITLE
Required Fields

### ISSUE THREAD

**Issue by Unleashed2k** [rscustom/rocksmith-custom-song-toolkit#202]
With the upcoming CustomsForge Song App, we've noticed two major problems that we're hoping you can assist us with. I've noticed the author field is not forced. Is there any way you can make it where the user will need to submit a username (it can be any alias they choose) in that area & save it along with the 2nd problem we have which is the version number is not stored.

My developers are working with the .json files inside the .psarc that start with "manifests/songs"

TL;DR we need the version and author saved within the files so we can identify if a song needs an update.

**Comment by zerkz** [comment]
An easy way to resolve above issue is just fork the project. Any changes to the case can be easily merged in from the base repo, and no one really has to worry about maintaining desired changes. Unless @fabianosan is ok with making the changes to the base repo for the sake of the song app.

**Comment by ray** [comment]
The author and version are already being written to the toolkit.version file.

**Comment by L0FKA** [comment]
One thing I actually think about is setup RSTK for the first run, ask user to fill required fields\show basic help\ some info etc

**Comment by cozy1** [comment]
This issue of "forcing author field" to be completed was discussed by devs when it was first implemented.  The perception is that this is big brother and users will not fill it in, or will fill in with garbage.  Suggest that on your end you provide a default value for when author field is empty.  If you prefer we could add "Unknown Author" as a default but then you risk that all CDLC will be created by "Unknown Author" because it well be left as default.  Flip of the coin.

**Comment by cozy1** [comment]
Closed due to inactivity.
